magento/magento2
# Add SMTP configuration + transport options

### Description (*)
This adds the SMTP Transport from Laminas, to configure external SMTP with authentication.

![image](https://user-images.githubusercontent.com/973269/108715485-3b09d900-751b-11eb-959e-d572df62d276.png)


### Fixed Issues (if relevant)
Fixes #26104
1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)
Create an SMTP account somewhere (Sendgrid, Mailgun, Mailtrap etc).
Go to System -> Mail Configuration and fill the details
Send any email (eg Customer Password reset)
See it works

### Questions or comments
- The XML Path constants are repeated a bit, perhaps it would be better to create a prefix path and call `$this->getConfig('host')`
- I disabled cram-md5 because it requires an external package. We could allow it and add a note to require it yourself if wanted.

### Contribution checklist (*)
